concept/type_/type_reader.rs

```diff
@@ -119,24 +172,53 @@ impl TypeReader {
 
     pub(crate) fn get_owns(
         snapshot: &impl ReadableSnapshot,
-        owner: impl OwnerAPI<'static>
-    ) -> Result<HashSet<Owns<'static>>, ConceptReadError>
-    {
+        owner: impl OwnerAPI<'static>,
+    ) -> Result<HashSet<Owns<'static>>, ConceptReadError> {
         let owns_prefix = build_edge_owns_prefix_from(owner.into_vertex());
         snapshot
             .iterate_range(KeyRange::new_within(owns_prefix, TypeEdge::FIXED_WIDTH_ENCODING))
             .collect_cloned_hashset(|key, _| {
                 let owns_edge = new_edge_owns(Bytes::Reference(key.byte_ref()));
-                Owns::new(ObjectType::new(owns_edge.from().into_owned()), AttributeType::new(owns_edge.to().into_owned())) // TODO: Should we make this more type safe.
+                Owns::new(
+                    ObjectType::new(owns_edge.from().into_owned()),
+                    AttributeType::new(owns_edge.to().into_owned()),
+                ) // TODO: Should we make this more type safe.
             })
             .map_err(|error| ConceptReadError::SnapshotIterate { source: error })
     }
 
-    pub(crate) fn get_plays(
+    pub(crate) fn get_owns_transitive<T>(
         snapshot: &impl ReadableSnapshot,
-        player: impl PlayerAPI<'static>
-    ) -> Result<HashSet<Plays<'static>>, ConceptReadError>
+        owner: T,
+    ) -> Result<HashMap<AttributeType<'static>, Owns<'static>>, ConceptReadError>
+    where
+        T: OwnerAPI<'static> + ReadableType<ReadOutput<'static> = T>, // ReadOutput=T is needed for supertype transitivity
     {
+        // TODO: Should the owner of a transitive owns be the declaring owner or the inheriting owner?
+        let mut transitive_owns: HashMap<AttributeType<'static>, Owns<'static>> = HashMap::new();
+        let mut overridden_owns: HashSet<AttributeType<'static>> = HashSet::new(); // TODO: Should this store the owns? This feels more fool-proof if it's correct.
+        let mut current_type = Some(owner);
+        while current_type.is_some() {
+            let declared_owns = Self::get_owns(snapshot, current_type.as_ref().unwrap().clone())?;
+            for owns in declared_owns.into_iter() {
+                let attribute = owns.attribute();
+                if !overridden_owns.contains(&attribute) {
+                    debug_assert!(!transitive_owns.contains_key(&attribute));
+                    transitive_owns.insert(owns.attribute(), owns.clone());
+                    if let Some(overridden) = Self::get_owns_override(snapshot, owns.clone())? {
+                        overridden_owns.add(overridden.attribute());
+                    }
+                }
+            }
+            current_type = Self::get_supertype(snapshot, current_type.unwrap())?;
+        }
+        Ok(transitive_owns)
+    }
```

Comment:
Logic to compute inherited owns. It's a bit of code but the logic should be the same for owns/plays and even relates.